### PR TITLE
Upgrade Radius on Kubernetes

### PR DESCRIPTION
**Pull request by Reshrahim**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [ ] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description
This PR covers Docs to upgrade Radius on Kubernetes 

### Summary
📄🛠️🔖

This pull request updates the installation section of the docs by deleting obsolete files, adding new content, and using a shortcode to include shared instructions. It also adds the front matter metadata for the section and an image for the overview page. It removes the unused `howto-rad-cli` and `howto-vscode` guides.

> _`read` shortcode helps_
> _install and upgrade Radius_
> _in autumn breeze_

### Walkthrough
*  Delete three files that are no longer relevant for the docs: `docs/content/guides/deploy-apps/howto-rad-cli/index.md`, `docs/content/installation/index.md`, and `docs/content/guides/author-apps/howto-vscode/index.md` ([link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-51471d0c7b00570203de4c451595228d4edba897989954843ce6b54e87b586cd), [link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-f4d8965d9ffd353505aaddfc3f121ee2dcb240df010fd47ebb792ddda010219e), [link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-2712d7e37701fd2b74aa9e8e5a856f17d290b33e5ead737b5d330a6929ed95f1))
* Add front matter metadata for the installation section of the docs ([link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-b34973f25da637b231056cdf7019b5c6195dc6de7c45686d7e4d3de16ead9f43L1-R6))
* Add content for the installation page of the docs, explaining how to install the rad CLI, the VS Code extension, and the Radius control-plane on Kubernetes ([link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-c2152f2547083976266b9cf0a6f932d6151e32f6538ff0de2f071cb65c49b4e4R1-R200))
* Add content for the overview page of the installation section, providing a brief introduction to the Radius platform and its components, and an image of the architecture ([link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-f3ac26864196de751f6aa144a228172a948fe57fb6481ed70993dde0baec44c0L1-R12))
* Add content for the upgrade page of the installation section, explaining how to upgrade Radius on Kubernetes by deleting and reinstalling the environments and control-plane ([link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-b698df6da33837f4459ef245a9686da47c8619a4584935670ba4e902b943f60aR1-R31))
* Add a shortcode definition for the `read` shortcode, which reads and prints the content of a file as markdown ([link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-99471be1631adaacae3e7867d71fc485d3cc4f2c1b96938dfd14572718f62f92L1-R3))
* Use the `read` shortcode to include a shared installation file in the installation, overview, and upgrade pages, to avoid repeating the same instructions ([link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-c2152f2547083976266b9cf0a6f932d6151e32f6538ff0de2f071cb65c49b4e4R1-R200), [link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-f3ac26864196de751f6aa144a228172a948fe57fb6481ed70993dde0baec44c0L1-R12), [link](https://github.com/radius-project/docs/pull/746/files?diff=unified&w=0#diff-b698df6da33837f4459ef245a9686da47c8619a4584935670ba4e902b943f60aR1-R31))

## Issue reference

[#713](https://github.com/radius-project/docs/issues/713)

### CODE REVIEW

File: docs/content/installation/upgrade/index.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (Reshrahim) [inline docs/content/installation/upgrade/index.md]: IMO the upgrade content best lands under the installation section. Currently the page on "Install Radius on Kubernetes section" is under the Kubernetes section in the operations guide. We could have an Installation section toc like below

Installation
 - Install Radius tools 
      - rad cli and VScode
 - Install Radius on Kubernetes
 - Upgrade Radius on Kubernetes 

Thoughts?

File: docs/content/_index.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ Radius extends strong existing tools to further streamline the app developer exp
       <a href="{{< ref getting-started >}}"><img src="vscode-logo.png" alt="Visual Studio Code logo" style="width:80%"></a>
     </td>
     <td style="width:50%;text-align:center">
-      <a href="{{< ref bicep >}}" target="_blank"><img src="bicep-logo.png" alt="Bicep logo" style="width:70%"></a>
+      <a href="{https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/}" target="_blank"><img src="bicep-logo.png" alt="Bicep logo" style="width:70%"></a>
```
Comment (AaronCrawfis) [inline docs/content/_index.md]:
```suggestion
      <a href="{https://learn.microsoft.com/azure/azure-resource-manager/bicep/}" target="_blank"><img src="bicep-logo.png" alt="Bicep logo" style="width:70%"></a>
```

URLs should be region-agnostic

File: docs/content/installation/upgrade/index.md (hunk 1 above)
Comment (AaronCrawfis) [reply docs/content/installation/upgrade/index.md]: A page like upgrading is a deep subject that involves planning, referencing supported versioning tables, and is a task that operators come to the docs for when planning their upgrade schedule. It's not something that the majority of users will need detailed docs on.

I'd recommend that this page instead goes under the operations guides under the Kubernetes section, where material like how to manage Kubnernetes clusters can go

File: docs/content/installation/_index.md
Comment (AaronCrawfis) [inline docs/content/installation/_index.md:1]: I like the idea of revamping the installation page. But could we please put that into its own PR that's separate from the upgrade page?

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]

File: docs/content/installation/upgrade/index.md (hunk 1 above)
Comment (Reshrahim) [reply docs/content/installation/upgrade/index.md]: Today we ask users to upgrade to the latest version as we continue to have breaking changes in every release. Once opensourced, all the existing Radius users will hit this page to upgrade to the latest open source release. 

I agree upgrading Radius is an operators job but we don't have versioning policy implemeted yet. Any user of Radius has to upgrade Radius to latest version to continue using all the latest features of Radius and we call out that as the first thing in our release emails as well. 

So thought it would be better to have the content in a place thats accesible and relatable to users under Installation instead of operations

Comment (Reshrahim) [reply docs/content/installation/upgrade/index.md]: Discussed in doc PR sync and agreed to keep it under Kubernetes section in operations

**Review by AaronCrawfis — APPROVED** [review]
